### PR TITLE
Project upgrade

### PR DESCRIPTION
**Pull request by taavi-valjaots**

### CODE REVIEW

File: shelltestrunner.cabal [hunk 1]
```diff
@@ -46,7 +46,7 @@ executable shelltest
     base                 >= 4     && < 5,
     Diff                 >= 0.2.0 && < 0.4,
     filemanip            >= 0.3   && < 0.4,
-    HUnit                            < 1.6,
+    HUnit                            < 1.7,
```
Comment (simonmichael) [inline shelltestrunner.cabal:49]: Allowing HUnit 1.6, I see. Does it compile with HUnit < 1.6 too ? We should probably keep that working. HUnit 1.6 isn't yet widespread, eg stackage LTS has 1.5.

File: Parse.hs [hunk 2]
```diff
@@ -11,15 +11,18 @@ import Types
...
-       case processed of Right text -> return (runParser p () fname text)
-                         (Left err)  -> return (Left err) -- To make haskell happy.
+parseFromFileWithPreprocessor p preproc fname =
+    do
+        h <- openFile fname ReadMode
+        hSetNewlineMode h universalNewlineMode
+        input <- hGetContents h
+        let processed = preprocess preproc input
+        case processed of    Right text -> return (runParser p () fname text)
+                             (Left err) -> return (Left err) -- To make haskell happy.
```
Comment (simonmichael) [inline Parse.hs:25]: I'm not sure what this means to me as a user - how would you explain more clearly in the manual or release notes ?

File: shelltestrunner.cabal (hunk 1 above)
Comment (simonmichael) [reply shelltestrunner.cabal:49]: Yes it seems fine with older HUnit too.

**Comment by simonmichael** [comment]
Oops I misplaced a comment in the merge commit message. Never mind.

**Comment by simonmichael** [comment]
And in case you don't see it, that comment was: 
Looks good! let's clarify the line ending change in changelog when you get a chance

File: Parse.hs (hunk 2 above)
Comment (taavi-valjaots) [reply Parse.hs:25]: Do You mean the line 25 or the way the line breaks are handled?

Comment (simonmichael) [reply Parse.hs:25]: I mean in general, what is the new functionality this provides.

Comment (taavi-valjaots) [reply Parse.hs:25]: For user it gives macros (-D) and this is documented in CHANGES. For developer it gives opportunity to develop some other preprocessors. Also it handles windows line breaks on unix platforms and this is also documented in CHANGES.

**Pull request merged**